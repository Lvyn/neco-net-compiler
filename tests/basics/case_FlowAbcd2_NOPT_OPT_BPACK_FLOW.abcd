net gate () :
    buffer state : str = "OPEN"  # 1-safe
    ([state-("OPEN"), state+("MOVING")] ;
     [state-("MOVING"), state+("CLOSED")] ;
     [state-("CLOSED"), state+("MOVING")] ;
     [state-("MOVING"), state+("OPEN")])
    * [False]

gate()
