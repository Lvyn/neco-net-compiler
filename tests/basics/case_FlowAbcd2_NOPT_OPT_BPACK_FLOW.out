[{
'gate().e' : [],
'gate().i#1' : [dot],
'gate().i#2' : [],
'gate().i#3' : [],
'gate().state' : ['MOVING'],
'gate().x' : [],
}, {
'gate().e' : [],
'gate().i#1' : [],
'gate().i#2' : [dot],
'gate().i#3' : [],
'gate().state' : ['CLOSED'],
'gate().x' : [],
}, {
'gate().e' : [dot],
'gate().i#1' : [],
'gate().i#2' : [],
'gate().i#3' : [],
'gate().state' : ['OPEN'],
'gate().x' : [],
}, {
'gate().e' : [],
'gate().i#1' : [],
'gate().i#2' : [],
'gate().i#3' : [dot],
'gate().state' : ['MOVING'],
'gate().x' : [],
}, ]